weaviate/collections/data.py

```diff
@@ -165,16 +180,17 @@ def _reference_add(self, from_uuid: UUID, from_property: str, ref: _Reference) -
                 raise UnexpectedStatusCodeException("Add property reference to object", response)
 
     def _reference_add_many(self, refs: List[DataReference]) -> BatchReferenceReturn:
-        return self._batch_rest.references(
-            [
-                _BatchReference(
-                    from_=f"{BEACON}{self.name}/{ref.from_uuid}/{ref.from_property}",
-                    to=f"{BEACON}{ref.to_uuid}",
-                    tenant=self._tenant,
+        batch: Deque[_BatchReference] = deque()
```

Comment:
why this change?

Reply:
So that the `.append` calls occur in `O(1)` time rather than `O(n)` time helping to improve the batching algo's performance

Reply:
you don't call .append in list comprehensions

did you do a benchmark to show that this is actually faster?

Reply:
(micro-benchmark of just that scenario is fine)

Reply:
We can change it to a `list` if you prefer but `.append` must still be called as the logic is different

Reply:
I see ok - I'd prefer if things like this would be their own PR. It is really hard to spot what belongs together and what doesn't and what the consequences of a change are.